src/lib.rs

```diff
@@ -471,6 +477,8 @@ where
 {
     // Current start position of next chunk
     cursor: usize,
+    // position of last DONTNEED fadvise
+    fadvise_cursor: usize,
```

Comment:
This seems more complicated than necessary.  Instead of using `WILLNEED`/`DONTNEED`, just use `FADV_SEQUENTIAL`, which tells the kernel to drop cached pages as soon as you've finished reading them, and to aggressively read ahead according to how much memory is available.  The kernel has better information than you do about this stuff, don't try to outsmart it.

Alternatively you could use `FADV_NOREUSE` to have the kernel automatically drop pages once you've read them.  But I still say `FADV_SEQUENTIAL` is the pedantically correct approach here.

Reply:
this is the write path. Will `FADV_SEQUENTIAL` do something useful on writes?

Reply:
I don't think `fadvise` does much at all for writes.